src/backend/distributed/metadata/node_metadata.c

```diff
@@ -234,6 +234,20 @@ citus_set_coordinator_host(PG_FUNCTION_ARGS)
 
 		UpdateNodeLocation(coordinatorNode->nodeId, nodeNameString, nodePort);
 
+		WorkerNode *updatedNode = FindWorkerNodeAnyCluster(nodeNameString, nodePort);
+		Assert(updatedNode != NULL);
+
+		if (EnableMetadataSync)
+		{
+			/* send the delete command to all primary nodes with metadata */
+			char *nodeDeleteCommand = NodeDeleteCommand(updatedNode->nodeId);
+			SendCommandToWorkersWithMetadata(nodeDeleteCommand);
+
+			/* send the insert command to all primary nodes with metadata */
+			char *nodeInsertCommand = NodeListInsertCommand(list_make1(updatedNode));
+			SendCommandToWorkersWithMetadata(nodeInsertCommand);
+		}
+
```

Comment:
I suppose this is lacking the fault-tolerance logic in citus_update_node, but seems like an important improvement over the status quo because otherwise we end up with inconsistent metadata.

(perhaps should even include in 11.3?)